tests/components/mqtt_eventstream/test_init.py

```diff
@@ -25,177 +23,170 @@ def mock_storage(hass_storage):
-        self.hass.block_till_done()
-
-        # Verify that the this entity was subscribed to the topic
-        mock_sub.assert_called_with(self.hass, sub_topic, ANY)
-
-    @patch("homeassistant.components.mqtt.async_publish")
-    @patch("homeassistant.core.dt_util.utcnow")
-    def test_state_changed_event_sends_message(self, mock_utcnow, mock_pub):
-        """Test the sending of a new message if event changed."""
-        now = dt_util.as_utc(dt_util.now())
-        e_id = "fake.entity"
-        pub_topic = "bar"
-        mock_utcnow.return_value = now
+async def add_eventstream(hass, sub_topic=None, pub_topic=None, ignore_event=None):
+    """Add a mqtt_eventstream component."""
+    config = {}
+    if sub_topic:
+        config["subscribe_topic"] = sub_topic
+    if pub_topic:
+        config["publish_topic"] = pub_topic
+    if ignore_event:
+        config["ignore_event"] = ignore_event
+    return await async_setup_component(
+        hass, eventstream.DOMAIN, {eventstream.DOMAIN: config}
+    )
 
-        # Add the eventstream component for publishing events
-        assert self.add_eventstream(pub_topic=pub_topic)
-        self.hass.block_till_done()
-
-        # Reset the mock because it will have already gotten calls for the
-        # mqtt_eventstream state change on initialization, etc.
-        mock_pub.reset_mock()
 
-        # Set a state of an entity
-        mock_state_change_event(self.hass, State(e_id, "on"))
-        self.hass.block_till_done()
-
-        # The order of the JSON is indeterminate,
-        # so first just check that publish was called
-        mock_pub.assert_called_with(self.hass, pub_topic, ANY)
-        assert mock_pub.called
-
-        # Get the actual call to publish and make sure it was the one
-        # we were looking for
-        msg = mock_pub.call_args[0][2]
-        event = {}
-        event["event_type"] = EVENT_STATE_CHANGED
-        new_state = {
-            "last_updated": now.isoformat(),
-            "state": "on",
-            "entity_id": e_id,
-            "attributes": {},
-            "last_changed": now.isoformat(),
-        }
-        event["event_data"] = {"new_state": new_state, "entity_id": e_id}
-
-        # Verify that the message received was that expected
-        result = json.loads(msg)
-        result["event_data"]["new_state"].pop("context")
-        assert result == event
-
-    @patch("homeassistant.components.mqtt.async_publish")
-    def test_time_event_does_not_send_message(self, mock_pub):
-        """Test the sending of a new message if time event."""
-        assert self.add_eventstream(pub_topic="bar")
-        self.hass.block_till_done()
-
-        # Reset the mock because it will have already gotten calls for the
-        # mqtt_eventstream state change on initialization, etc.
-        mock_pub.reset_mock()
+async def test_setup_succeeds(hass, mqtt_mock):
+    """Test the success of the setup."""
+    assert await add_eventstream(hass)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)